pkg/platform/model/svc.go

```diff
@@ -133,3 +150,20 @@ func (m *SvcModel) ConfigChanged(ctx context.Context, key string) error {
 
 	return nil
 }
+
+func jsonFromMap(m map[string]interface{}) string {
+	d, err := json.Marshal(m)
+	if err != nil {
+		return fmt.Sprintf("cannot marshal map (%q) as json: %v", stringFromMap(m), err)
+	}
+	return string(d)
+}
+
+func stringFromMap(m map[string]interface{}) string {
+	var s, sep string
+	for k, v := range m {
+		s += fmt.Sprintf("%s%s: %#v", sep, k, v)
+		sep = ", "
+	}
+	return s
+}
```

Comment:
Doesn't look like these are being used?

Reply:
These are used to make the debugging output more human readable - json first, then a basic string fallback.